src/Data/Array/Polarized.hs

```diff
@@ -129,7 +129,9 @@ import Data.Vector (Vector)
 -- | Convert a pull array into a push array.
 -- NOTE: this does NOT require allocation and can be in-lined.
 transfer :: Pull.Array a %1-> Push.Array a
-transfer (Pull.Array f n) = Push.Array (\g -> DArray.fromFunction (\i -> g (f i))) n
+transfer (Pull.Array f n) =
+  Push.Array
+    (\k -> Prelude.foldl (\m i -> m <> (k (f i))) mempty [0..(n-1)])
```

Comment:
I'm not sure, but in my experience `foldl` is rarely the most efficient fold. Probably I'd go with `foldl'`.

Reply:
We can't use a linear `foldMap` without changing the type of `Push` and `Pull` to hold `Int %1-> a` and `(a %1-> m) -> m` respectively. I used `Prelude.foldMap`.

Reply:
Or did you mean a `foldMap` on `Pull` arrays? @aspiwack

Reply:
I meant `foldMap` on lists. But you are right that there is some tension in the types. Let's forget about this for now.